lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -361,8 +377,52 @@ export class PublicClientApplication {
         // block the reload if it occurred inside a hidden iframe
         BrowserUtils.blockReloadInHiddenIframes();
 
-        // Send request to renew token. Auth module will throw errors if token cannot be renewed.
-        return await this.authModule.getValidToken(tokenRequest);
+        try {
+            // Send request to renew token. Auth module will throw errors if token cannot be renewed.
+            return await this.authModule.getValidToken(tokenRequest);
+        } catch (e) {
+            const isServerError = e instanceof ServerError;
+            const isInvalidGrantError = (e.errorCode === BrowserConstants.INVALID_GRANT_ERROR);
+            if (isServerError && isInvalidGrantError) {
```

Comment:
Could there be a `invalidGrant` error that isn't a server error?

Reply:
If not, I might consider doing a `switch` statement here, in case we need to special case handle other error types in the future.

Reply:
It is possible, but I don't think it should be a case we consider. By using the class distinction we can short-circuit the check so that if an error other than a ServerError is thrown (i.e. configuration or ClientAuthError), we immediately throw rather than checking the auth code.

Reply:
It's my understanding that the ```invalid_grant``` error is equivalent to Interaction Required. Would this not be a case we would want to throw?

Reply:
For now we will use silent handler to check if there is still a valid session. It should still throw an interaction required error. We will update this when we start receiving a bad_token error.